### PR TITLE
Implement metarange iterator (atop sstables) and metarange managers

### PR DESCRIPTION
**Pull request by arielshaqed**
- Implement iterator/reader (`committed/iterator.go` and friends)
- Move metarange manager interface to `meta_range.go`
- Implement metarange manager
- Apply large amount of duct tape to  make tests pass

### CODE REVIEW

File: graveler/committed/range_manager.go [hunk 1]
```diff
@@ -40,6 +40,10 @@ type RangeManager interface {
 	// found, it return (nil, ErrNotFound).
 	GetValue(ns Namespace, id ID, key Key) (*Record, error)
 
+	// GetValueGE returns the first value keyed at or after key in the Range referenced by
+	// id.  If all values are keyed before key, it returns (nil, ErrNotFound).
+	GetValueGE(ns Namespace, id ID, key Key) (*Record, error)
```
Comment (nopcoder) [inline graveler/committed/range_manager.go:45]: Still looks like a helper function that uses NewRangeIterator

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Ready for another round.  I added *tests* for `SeekGE` on iterators, everything broke, but at least I think now it's slightly easier to understand again.

PTAL...

**Review by nopcoder — APPROVED** [review]

File: graveler/committed/iterator_test.go [hunk 2]
```diff
@@ -66,7 +105,24 @@ func keysByRanges(t testing.TB, it committed.Iterator) []rangeKeys {
...
+	return c
+}
+
+// succ returns the key right after k.
+func succ(k committed.Key) committed.Key {
+	c := make(committed.Key, len(k)+1)
+	copy(c, k)
+	c[len(c)] = 0
+	return c
+}
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply graveler/committed/iterator_test.go:114]: I don't think lexicographical ordering works like that.  The next all-digits string after "100" is "1000" not "101".

Put differently, any string with prefix k comes before the string you get from adding 1 to that prefix in base-256.  So the first string after k is k followed by the first non-empty string, which is "0" (in base-10).

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Pulling once I get tests etc.

**Comment by arielshaqed** [comment]
Pulling without waiting for lgtm.com because (a) previous commits passed; (b) it's really slow today; (c) we do get reports on `master`.

Thanks!

**Pull request merged**